sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuExpressions.scala

```diff
@@ -148,6 +149,35 @@ trait CudfUnaryExpression extends GpuUnaryExpression {
   override def doColumnar(input: GpuColumnVector): ColumnVector = input.getBase.unaryOp(unaryOp)
 }
 
+trait GpuRoundBase extends CudfRoundBase with GpuExpression {
+  def roundMode : RoundMode
+  def doColumnar(val0: GpuColumnVector, val1: Scalar): ColumnVector
+
+  override def columnarEval(batch: ColumnarBatch): Any = {
+    var lhs: Any = null
+    var rhs: Any = null
+    try {
+      lhs = left.columnarEval(batch)
+      rhs = right.columnarEval(batch)
+
+      (lhs, rhs) match {
+        case (l: GpuColumnVector, r) =>
+          withResource(GpuScalar.from(r, right.dataType)) { scalar =>
+            GpuColumnVector.from(doColumnar(l, scalar), dataType)
+          }
+        case _ => null
+      }
+    } finally {
+      if (lhs.isInstanceOf[AutoCloseable]) {
+        lhs.asInstanceOf[AutoCloseable].close()
+      }
+      if (rhs.isInstanceOf[AutoCloseable]) {
```

Comment:
`rhs` could be null at this point so needs a null check here. Perhaps it would be better to introduce something similar to the `withResource` method that can work with `Any`'s that are possibly also `AutoCloseable`?

Reply:
I added on https://github.com/NVIDIA/spark-rapids/blob/cce1ec1ed401bcd7465f5bef9ab28ed38aa5a9a8/sql-plugin/src/main/scala/com/nvidia/spark/rapids/Arm.scala#L52-L62

Reply:
It turns out we already have a `withResource` that can work here. 

```scala
withResourceIfAllowed(left.columnarEval(batch)) { lhs =>
  withResourceIfAllowed(right.columnarEval(batch)) { rhs =>
    (lhs, rhs) match {
      case (l: GpuColumnVector, r) =>
        withResource(GpuScalar.from(r, right.dataType)) { scalar =>
          GpuColumnVector.from(doColumnar(l, scalar), dataType)
        }
      case _ => null
    }
  }
}
```

Reply:
But further to the point why is this not a `GpuBinaryExpression`?  `BRound` and `Round` are both `BinaryExpression`s which would clean up this code.

Reply:
Thanks @andygrove  and @revans2 for taking a look.  Sorry missed this conversation. The reason I was not making it as `GpuBinaryExpression` is because there isn't an BinaryOp for round or BRound in JNI or in cudf binaryops.  And I was not sure what would be the overidden enum be for these operators. But this also isn't working for all cases. So I will try how to go about it as a `GpuBinaryExpression`

Reply:
`GpuBinaryExpression` does not assume an enum but `CudfBinaryExpression` does.

Reply:
Thanks! changed it to use `GpuBinaryExpression`. Will be in next patch.